### ISSUE TITLE
Pre-train model for evaluation?

### ISSUE THREAD

**Issue by Minotaur-CN** [oandrienko/fast-semantic-segmentation#2]
Dear oandrienko,

Thanks for your code , 
1. Could you upload your trained model for evaluation? 
2.  Inference.py:100, output_channels is not defined in code. If setting output_channels=1, the gray result was showed in the result. If setting output_channels!=1. raise TypeError("Cannot handle this data type")  on line 102 "im = Image.fromarray(predictions)"

Thanks

**Comment by oandrienko** [comment]
Hey, thanks for your interest in the project and sorry for the delay.

Thanks for brining up the issue with the inference script. I have updated it so you can now use it with both PSPNet50 or an ICNet model to output either RGB colours or Cityscapes Label IDs. 

For the pre-trained ICNet model: I have been away for the last while but am now in the process of figuring out the optimal training and compression setup for ICNet. I can mention you in this issue once I have the ICNet weights uploaded in the near future. In the meantime, I do have the baseline PSPNet50 model up which I have recently uploaded. You can find a link to the trained PSPNet model [here](https://github.com/oandrienko/fast-semantic-segmentation/blob/master/docs/model_zoo.md). There is also a usage guide for training PSPNet50 [here](https://github.com/oandrienko/fast-semantic-segmentation/blob/master/docs/pspnet.md).

**Comment by Minotaur-CN** [comment]
Thanks for your reply!

**Comment by oandrienko** [comment]
Hey @Minotaur-CN, I recently uploaded an older compressed ICNet model I trained back in September. If can be found [here](https://github.com/oandrienko/fast-semantic-segmentation/blob/master/docs/model_zoo.md). A small guide for training can be found [here](https://github.com/oandrienko/fast-semantic-segmentation/blob/master/docs/icnet.md).

The model is ~1.5% mIoU off from the original accuracy reported in the paper. I haven't had much time but am still working on improving the training process. Hope this helps.
